apiserver/facades/agent/credentialvalidator/state.go

```diff
@@ -18,21 +17,10 @@ type ModelAccessor interface {
 	WatchModelCredential() state.NotifyWatcher
 }
 
-// StateAccessor exposes State methods needed by credential validator.
-type StateAccessor interface {
-	credentialcommon.StateBackend
-	Model() (ModelAccessor, error)
-}
-
 type stateShim struct {
 	*state.State
 }
 
-// Model returns model from this shim.
-func (s *stateShim) Model() (ModelAccessor, error) {
-	return s.State.Model()
-}
-
 func (s *stateShim) CloudCredentialTag() (names.CloudCredentialTag, bool, error) {
```

Comment:
We should be able to delete this entire file

Reply:
Hi ian, while refactoring this I realised we are using InvalidateModelCredential. We technically would still need the backend right?

https://github.com/CodingCookieRookie/juju/blob/JUJU-7863-remove-basic-info-agent-credential-validator/api/agent/credentialvalidator/credentialvalidator.go#L86-L99

Reply:
Yup I have removed that part from the credentialvalidator agent facade, but what I meant was we can't delete the entire file right? Since we would still need the implementation for CloudCredentialTag(). Perhaps I could change stateShim to take in modelInfo instead of state for CloudCredentialTag()?

Reply:
`watch, err := api.credentialService.WatchCredential(ctx, credential.KeyFromTag(credentialTag))`
We don't need tag. Or `KeyFromTag`
We have everything we need to create the key from ModelInfo